File: remote/src/main/java/com/kelsos/mbrc/annotations/PlayerAction.java
package com.kelsos.mbrc.annotations;

import android.support.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class PlayerAction {
  public static final String STOP = "stop";
  public static final String PLAY = "play";
  public static final String PAUSE = "pause";
  public static final String NEXT = "next";
  public static final String PREVIOUS = "previous";
  public static final String PLAY_PLAUSE = "playpause";

  private PlayerAction() {
    //no instance
  }

  @Retention(RetentionPolicy.SOURCE)
  @StringDef({
                 NEXT,
                 STOP,
                 PLAY,
                 PAUSE,
                 PREVIOUS,
                 PLAY_PLAUSE
             })
  public @interface Action {
  }
}

File: mbrc-lib/src/main/kotlin/com/kelsos/mbrc/annotations/PlayerAction.kt
package com.kelsos.mbrc.annotations

import android.support.annotation.StringDef

object PlayerAction {
    const val STOP = "stop"
    const val PLAY = "play"
    const val PAUSE = "pause"
    const val NEXT = "next"
    const val PREVIOUS = "previous"
    const val PLAY_PLAUSE = "playpause"

    @kotlin.annotation.Retention(AnnotationRetention.SOURCE)
    @StringDef(NEXT, STOP, PLAY, PAUSE, PREVIOUS, PLAY_PLAUSE)
    annotation class Action
}
